application/src/main/java/org/thingsboard/server/controller/DashboardController.java

```diff
@@ -107,17 +108,61 @@ public Dashboard getDashboardById(@PathVariable(DASHBOARD_ID) String strDashboar
         }
     }
 
+    @PreAuthorize("hasAnyAuthority('TENANT_ADMIN', 'CUSTOMER_USER')")
+    @RequestMapping(value = "/dashboard/{dashboardId}/entityConfig/{entityConfigId}/restore", method = RequestMethod.POST)
+    @ResponseBody
+    public Dashboard restoreDashboardConfig(@PathVariable(DASHBOARD_ID) String strDashboardId,
+                                            @PathVariable("entityConfigId") String strEntityConfigId,
+                                            @RequestParam(required = false) String comment) throws ThingsboardException {
+        checkParameter(DASHBOARD_ID, strDashboardId);
+        checkParameter("entityConfigId", strEntityConfigId);
+        try {
+            DashboardId dashboardId = new DashboardId(toUUID(strDashboardId));
+            checkDashboardId(dashboardId, Operation.READ);
```

Comment:
So, the user with read-only permissions can restore an old copy of the dashboard?